src/evaluate.cpp

```diff
@@ -887,14 +889,9 @@ namespace {
 #ifdef THREECHECK
             if (pos.is_three_check())
             {
-                switch(pos.checks_given(Them))
-                {
-                case CHECKS_NB:
-                case CHECKS_3:
-                case CHECKS_2:  kingDanger += 2 * kingDanger; break;
-                case CHECKS_1:  kingDanger += kingDanger; break;
-                case CHECKS_0:  kingDanger += kingDanger / 2; break;
-                }
+              int i = std::max(0, 2 - pos.checks_given(Them));
+              if (i < 3)
```

Comment:
It might make sense to support N-check, but as long as the rest of the code does not support it, IMO it does not make sense to take account of it at this point.